lib/client/api_confluence.es6.js

```diff
@@ -0,0 +1,191 @@
+      documentation: `A DAO that contains number of aggressive removal APIs for
+          each browser associated with date.`,
+      typeName: 'aggressiveRemovalDataDAO',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'mlang',
+      documentation: 'The mlang expressions singleton.',
+      factory: function() {
+        return this.ExpressionsSingleton.create();
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'getApiVelocity',
+      documentation: `Read from apiVelocityDAO and returns API velocity
+          metric for each browser.`,
+      returns: {
+        typeName: 'JSON',
+        documentation: `JSON of the form:
+            {browserName: [
+              {
+                releaseDate,
+                browserName,
+                browserVersion,
+                osName,
+                osVersion,
+                currBrowser: The browser object which is released at
+                    the releaseDate,
+                prevBrowser: <browser object of previous release>,
+                totalApis: <total number of APIs>,
+                newApis: <number of new APIs>,
+                removedApis: <number of removed APIs>,
+              }, ...
+            ], ...}`,
+      },
+      code: function() {
+        return this.apiVelocityDAO.select(this.mlang.GROUP_BY(
+            this.ApiVelocityData.BROWSER_NAME,
+            this.ArraySink.create())).then((groups) => {
+          let apiVelocityResult = {};
+          for (let i = 0; i < groups.groupKeys.length; i++) {
+            let browserName = groups.groupKeys[i];
+            apiVelocityResult[browserName] =
+              groups.groups[browserName].a.map((apiVelocityData) => {
+                return {
+                  releaseDate: apiVelocityData.releaseDate,
+                  browserName: apiVelocityData.browserName,
+                  browserVersion: apiVelocityData.currBrowser.browserVersion,
+                  osName: apiVelocityData.currBrowser.osName,
+                  osVersion: apiVelocityData.currBrowser.osVersion,
+                  totalApis: apiVelocityData.totalApis,
+                  newApis: apiVelocityData.newApis,
+                  removedApis: apiVelocityData.removedApis,
+                  prevBrowser: apiVelocityData.prevBrowser,
+                  currBrowser: apiVelocityData.currBrowser,
+                };
+            });
+          }
+          return apiVelocityResult;
+        });
+      },
+    },
+    {
+      name: 'getFailureToShip',
+      documentation: `Read from failure to ship DAO and return failure
+          to ship metrics metric for each browser.`,
+      returns: {
+        typeName: 'JSON',
+        documentation: `JSON of the form:
+            {browserName: [
+              [FailureToShipData, ...]
+            ], ...}`,
+      },
+      code: function() {
+        return this.failureToShipDAO.orderBy(this.FailureToShipData.DATE
+            ).select(this.mlang.GROUP_BY(
```

Comment:
Bracket should be on the previous line, or the linebreak should be somewhere else.